constph.py

```diff
@@ -573,14 +618,14 @@ def setTitrationState(self, titration_group_index, titration_state_index, contex
             # TODO: Handle Custom forces, looking for "charge" and "chargeProd".
             for (charge_index, atom_index) in enumerate(atom_indices):
                 if force_classname == 'NonbondedForce':
-                    [charge, sigma, epsilon] = force.getParticleParameters(atom_index)
-                    if debug: print (" modifying NonbondedForce atom %d : (charge, sigma, epsilon) : (%s, %s, %s) -> (%s, %s, %s)" % (
-                    atom_index, str(charge), str(sigma), str(epsilon), str(charges[charge_index]), str(sigma), str(epsilon)))
+                    [charge, sigma, epsilon] = map(strip_in_unit_system, force.getParticleParameters(atom_index))
+                    # if debug: print (" modifying NonbondedForce atom %d : (charge, sigma, epsilon) : (%s, %s, %s) -> (%s, %s, %s)" % (
+                    # atom_index, str(charge), str(sigma), str(epsilon), str(charges[charge_index]), str(sigma), str(epsilon)))
                     force.setParticleParameters(atom_index, charges[charge_index], sigma, epsilon)
                 elif force_classname == 'GBSAOBCForce':
-                    if debug: print (" modifying GBSAOBCForce atom %d : (charge, radius, scaleFactor) : (%s, %s, %s) -> (%s, %s, %s)" % (
-                    atom_index, str(charge), str(radius), scaleFactor, str(charges[charge_index]), str(radius), scaleFactor))
-                    [charge, radius, scaleFactor] = force.getParticleParameters(atom_index)
+                    # if debug: print (" modifying GBSAOBCForce atom %d : (charge, radius, scaleFactor) : (%s, %s, %s) -> (%s, %s, %s)" % (
+                    # atom_index, str(charge), str(radius), scaleFactor, str(charges[charge_index]), str(radius), scaleFactor))
+                    [charge, radius, scaleFactor] = map(strip_in_unit_system, force.getParticleParameters(atom_index))
```

Comment:
We want to avoid calling `getParticleParameters` by caching `[charge, radius, scaleFactor]`.